libs/ui/dataviz/src/lib/table/table.component.stories.ts

```diff
@@ -29,29 +34,89 @@ export default {
       ],
     }),
     componentWrapperDecorator(
-      (story) => `<div style="max-width: 700px; height: 400px">${story}</div>`
+      (story) => `<div style="height: 400px">${story}</div>`
     ),
   ],
 } as Meta<TableComponent>
 
+export class MockBaseReader extends BaseFileReader {
+  override getData(): Promise<{
+    items: DataItem[]
+    properties: PropertyInfo[]
+  }> {
+    return Promise.resolve({
+      items: [
+        {
+          type: 'Feature',
+          geometry: null,
+          properties: {
+            id: '0001',
+            firstName: 'John',
+            lastName: 'Lennon',
+          },
+        },
+        {
+          type: 'Feature',
+          geometry: null,
+          properties: {
+            id: '0002',
+            firstName: 'Ozzy',
+            lastName: 'Osbourne',
+          },
+        },
+        {
+          type: 'Feature',
+          geometry: null,
+          properties: {
+            id: '0003',
+            firstName: 'Claude',
+            lastName: 'François',
+          },
+        },
+      ],
+      properties: [
+        { name: 'id', label: 'id', type: 'string' },
+        { name: 'firstName', label: 'Firstname', type: 'string' },
+        { name: 'lastName', label: 'Lastname', type: 'string' },
+      ],
+    })
+  }
+  // override read(): Promise<DataItem[]> {
+  //   return Promise.resolve([
+  //     {
+  //       type: 'Feature',
+  //       geometry: null,
+  //       properties: {
+  //         id: '0001',
+  //         firstName: 'John',
+  //         lastName: 'Lennon',
+  //       },
+  //     },
+  //     {
+  //       type: 'Feature',
+  //       geometry: null,
+  //       properties: {
+  //         id: '0002',
+  //         firstName: 'Ozzy',
+  //         lastName: 'Osbourne',
+  //       },
+  //     },
+  //     {
+  //       type: 'Feature',
+  //       geometry: null,
+  //       properties: {
+  //         id: '0003',
+  //         firstName: 'Claude',
+  //         lastName: 'François',
+  //       },
+  //     },
+  //   ])
+  // }
+}
+const reader = new MockBaseReader('')
+
 export const Primary: StoryObj<TableComponent> = {
   args: {
-    data: [
-      {
-        id: '0001',
-        firstName: 'John',
-        lastName: 'Lennon',
-      },
-      {
-        id: '0002',
-        firstName: 'Ozzy',
-        lastName: 'Osbourne',
-      },
-      {
-        id: '0003',
-        firstName: 'Claude',
-        lastName: 'François',
-      },
-    ],
+    dataset: reader,
```

Comment:
how about reading a stable dataset somewhere, such as https://france-geojson.gregoiredavid.fr/repo/departements.geojson ?

Reply:
Hum, I was wondering if it's a good idea to create a dependence to an external data source.